### ISSUE TITLE
SSL I/O Error when fetching mail

### ISSUE THREAD

**Issue by dalb8** [k9mail/k-9#1993]
Ever since updating to 5.201 (or 5.200; all was fine on 5.200-RC anyhow) there's a problem fetching email. It constantly downloads data, takes ages and then says "SSL exception:write error" under the inbox. I just installed 5.202 but it seems the problem is still there.

**Comment by philipwhiuk** [comment]
If you enable debugging and look in the k9mail-errors folder is there a corresponding 'mail'. If so can you provide the content of that.

**Comment by dalb8** [comment]
K9-Mail version: 5.202
Device make: LGE
Device model: LG-H815
Android version: 6.0

com.fsck.k9.mail.MessagingException: IO Error
at com.fsck.k9.mail.store.imap.ImapFolder.ioExceptionHandler(ImapFolder.java:1366)
at com.fsck.k9.mail.store.imap.ImapFolder.appendMessages(ImapFolder.java:1223)
at com.fsck.k9.controller.MessagingController.processPendingAppend(MessagingController.java:1869)
at com.fsck.k9.controller.MessagingController.processPendingCommandsSynchronous(MessagingController.java:1744)
at com.fsck.k9.controller.MessagingController.synchronizeMailboxSynchronous(MessagingController.java:779)
at com.fsck.k9.controller.MessagingController$36.run(MessagingController.java:4010)
at com.fsck.k9.controller.MessagingController.runInBackground(MessagingController.java:204)
at com.fsck.k9.controller.MessagingController.access$000(MessagingController.java:115)
at com.fsck.k9.controller.MessagingController$1.run(MessagingController.java:173)
at java.lang.Thread.run(Thread.java:818)
Caused by: javax.net.ssl.SSLException: Write error: ssl=0x7f88826600: I/O error during system call, Broken pipe
at com.android.org.conscrypt.NativeCrypto.SSL_write(Native Method)
at com.android.org.conscrypt.OpenSSLSocketImpl$SSLOutputStream.write(OpenSSLSocketImpl.java:771)
at java.io.BufferedOutputStream.write(BufferedOutputStream.java:174)
at java.io.FilterOutputStream.write(FilterOutputStream.java:129)
at com.fsck.k9.mail.filter.EOLConvertingOutputStream.writeByte(EOLConvertingOutputStream.java:61)
at com.fsck.k9.mail.filter.EOLConvertingOutputStream.write(EOLConvertingOutputStream.java:31)
at java.io.FilterOutputStream.write(FilterOutputStream.java:114)
at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:1793)
at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:1769)
at org.apache.commons.io.IOUtils.copy(IOUtils.java:1744)
at com.fsck.k9.mailstore.FileBackedBody.writeTo(FileBackedBody.java:45)
at com.fsck.k9.mail.internet.MimeBodyPart.writeTo(MimeBodyPart.java:147)
at com.fsck.k9.mail.internet.MimeMultipart.writeTo(MimeMultipart.java:93)
at com.fsck.k9.mail.internet.MimeMessage.writeTo(MimeMessage.java:422)
at com.fsck.k9.mailstore.LocalMessage.writeTo(LocalMessage.java:571)
at com.fsck.k9.mail.store.imap.ImapFolder.appendMessages(ImapFolder.java:1169)
... 8 more

... and 3 other similar messages

**Comment by kfomichev** [comment]
Hello!
I also have SSL related issue since 3-4 January. I don't know if there was and app update or not. But I guess there was.
I'm trying to establish SSL encrypted IMAP session . My server is an old MDaemon server, which doesn't support modern encryption ciphers. The server reports SSL negotiation failure, and this failure has known reason - the server only supports RC4 encryption with MD5 authorization, it's considered to be insecure,  Modern ciphers use AES/SHA. Like TLS_RSA_WITH_AES_128_CBC_SHA AES128-SHA . This in unsolvable at my server side.
Could anybody confirm that latest k9 Mail has changed negotiation cipher? If yes, could anybody consider implementing a user-configurable fallback to less secure ciphers?

**Comment by philipwhiuk** [comment]
Yes 5.2x updated the list of allowed ciphers. 

I think we will probably support fallback in future, but only after we've implemented better UX to show what's being used / fallen back to ( #893 ). Ideally it will be a bit like Chrome's warning screen.

**Comment by philipwhiuk** [comment]
To me the original error here looks like a connectivity problem with the server. It looks like the server is closing our connection, but there's no info why.

**Comment by kfomichev** [comment]
Ok, thank you. Maybe original error is really unrelated to my, in my log I see javax.net.ssl.SSLHandshakeexception

**Comment by dalb8** [comment]
It's a Yahoo account btw

**Comment by dalb8** [comment]
I did 'recreate data' on the account and it seemed to work OK for about a day before the error reocurred.

**Comment by dalb8** [comment]
The error has disappeared with 5.203.

**Comment by Valodim** [comment]
Yay. Guess the broken pipe was caused by some exception on the other side that was fixed in another issue? Oh well, I won't complain about an issue resolving itself :)
